dashboard/app/views/lessons/show.html.haml

```diff
@@ -2,13 +2,15 @@
 
 %link{href: asset_path('css/lessons.css'), rel: 'stylesheet', type: 'text/css'}
 %script{src: webpack_asset_path('js/lessons/show.js'),
-        data: {lesson: @lesson_data.to_json}}
+        data: {lesson: @lesson_data.to_json, customBlocksConfig: Block.for('GamelabJr').to_json}}
+%script{src: webpack_asset_path('js/blockly.js')}
```

Comment:
On the topic of google blockly, it'll be more difficult to figure out which blockly to use in lessons (though an individual will rarely embed blocks from multiple labs). In the code rollup page for scripts, it might be nearly impossible (blocks will be embedded from multiple labs). Not really sure the way to reconcile this, so open to suggestions.

Reply:
interesting... i'm not positive this is a good suggestion, but maybe that means we should load CDO blockly on these pages until all labs are on Google blockly? our goal is to stay compatible with both versions and i think the only user-facing difference should be some small UI changes between CDO and Google blockly. cc'ing @katleiahramos and @ajpal in case i'm totally off here

Reply:
That makes sense to me! Anjali and I were talking about using Google blockly here and realized there's actually a bit more work to make that work (as it stands, the page crashes if we try to load Google blockly....). I'd rather look into that as a follow up though

Reply:
> maybe that means we should load CDO blockly on these pages until all labs are on Google blockly? our goal is to stay compatible with both versions and i think the only user-facing difference should be some small UI changes between CDO and Google blockly.

This sounds right to me if these assumptions are correct. I'm super (duper!) excited about the blockly upgrades, but I'd also like to stay out of lockstep as each lab is upgraded to Google Blockly (and/or would like to make sure we have a way to make the rollup pages work at all in the first place). ideally it would work something like this:
1. lesson plans use CDO blockly everywhere
2. wait until all labs are upgraded to Google Blockly
3. update lesson plans to only use Google Blockly

Reply:
I think that approach sounds reasonable, at least for now. The only issue I can see is if we start using some of the new Google Blockly features in some labs before all the labs are upgraded. If, for example, we start using the slider field in Sprite Lab before we are fully finished with the migration, then we'd either have to implement the slider field in Cdo Blockly (which would sort of be wasted effort) or make sure these docs pages would use Google Blockly for Sprite lab blocks at that point

Reply:
Courses D and E look like a good forcing function for dealing with this problem. if I am reading correctly, it looks like they contain the following lab types: 
* **Course D: Artist, Studio/playlab, Bounce, Craft, Dancelab, Karel, Maze**
* **Course E: Artist, Craft, GamelabJr/spritelab, Karel**

if we take the "no dessert until vegetables are eaten" approach, we could say that we do not get to use sliders, etc in spritelab until all of these lab types are upgraded to Google Blockly. @KylieModen how does that sound? cc @tess323 for PM discussion. if that does NOT sound good, then we should be ready for some eng work (TBD) or user pain (regression in code block rendering) if/when we start using new features like sliders.

or if we are going or a more surgical approach, and the sliders are going to be used in spritelab but not dance, it's possible we could start using them in spritelab once only the Course E lab types have all been upgraded.

sorry to get into the weeds here!

Reply:
No this is a great conversation! What's also interesting here is that only spritelab uses code docs, so, currently, we're not blocked on say Artist moving to Google blockly. However, if any other lab wants to start using code docs, we may hit a problem. I keep coming back to Poetry in my head, which is built on Google blockly and is new. I don't think there are any plans to create code docs for it but I definitely don't know for sure and I don't know if we could get into a sticky spot if we decided to

Reply:
Dave and I chatted offline and we think it would be clearest if we say that Google blockly won't be used for code docs until Spritelab is using Google blockly. This gives us time to figure out any issues, plus some Spritelab blocks are defined in the Google blockly repo and I _think_ those would break if we started using Google blockly.

I am making the assumption that, for each lab, it is either completely using Google blockly or completely using the blockly fork, so let me know if that's wrong. But as Spritelab is the only lab *currently* using code docs, this means that we're tied to Spritelab moving to Google blockly. Does this make sense?

Reply:
That all seems right to me, except I'm not sure I understand this part: "some Spritelab blocks are defined in the Google blockly repo and I think those would break if we started using Google blockly" Which blocks are you talking about there?

Reply:
The blocks in [this directory](https://github.com/code-dot-org/code-dot-org/tree/staging/dashboard/config/programming_expressions/spritelab) that begin with `codestudio_`. When I looked for their definitions, I found them in the blockly fork but now I'm realizing they are likely also defined in the Google blockly repo.... 

As an example, `codestudio_ifStatement` is defined [here](https://github.com/code-dot-org/blockly/blob/7de3f7b4845b506cfa2c951a2fabcc1a956fd53e/blocks/logic.js#L189) in the blockly fork but I think [this](https://github.com/google/blockly/blob/423d2e58a13620c255532a24f8005707bad9d7b6/blocks/logic.js#L234) is the version in Google blockly.

Reply:
Ohh got it. Yeah there's a set of core blocks that come as default in Blockly, and I think they're probably the same (or at least very similar) between versions. I think they'll probably "just work" when we switch spritelab to Google blockly, or if not, would probably relatively easy to fix

Reply:
oops, ok, so based on that, it sounds like the work in this PR won't actually put us into a situation where we can't use new Google Blockly features, because today there is only one lab type (spritelab) on any code docs rollup page. but (per offline convo with Bethany) if we did want to add block docs for another lab type, then we might get into that constrained situation. To try and simplify this decision tree, I would suggest the following strategy:
1. no adding code docs for new lab types for now (or if we do, no rendering its blockly blocks)
2. upgrade spritelab to Google Blockly
3. update lesson plans to (only) render Google Blockly
4. open season to add code docs for new lab types, but only if they are using Google Blockly

a situation I would want to avoid is, adding code docs for more lab types on CDO Blockly, making it hard to upgrade the lesson plan / rollup pages to use Google Blockly, making it harder for spritelab to start using new Google Blockly features. other strategies could work too.

Reply:
that plan sounds good to me. cc'ing @aaronwaggener just so he's aware of this dependency as we transition more labs to google blockly!

Reply:
This course of action makes sense given our current status with moving to Google Blockly. I will callout that while we plan to move Sprite Lab and want to move everything to Google Blockly, there has been some discussion around deprecation/sunsetting of Play Lab and as such possibly not migrating Play Lab to Google Blockly. Something to keep in mind as we discuss in future.

Reply:
^ Kylie definitely knows best here so I defer to her judgment.  Thanks for the heads up!

Reply:
Thanks all for all the discussion! One last callout is that all the code I've written is based off of the logic we use for https://levelbuilder-studio.code.org/pools/GamelabJr/blocks, so when Spritelab moves to Google blockly, this page may need the same updates as lesson plans/code docs.